:100644 100644 14f6ec8... 0000000... M	dcdis.c
:100644 100644 d385f3b... 0000000... M	dcdis.h
:100644 100644 bc1039d... 0000000... M	decode.c
:100644 100644 48d01e0... 0000000... M	decode.h
:100644 100644 6186d00... 0000000... M	symtab.c
:100644 100644 4dc82cc... 0000000... M	symtab.h

diff --git a/dcdis.c b/dcdis.c
index 14f6ec8..c4bafcb 100644
--- a/dcdis.c
+++ b/dcdis.c
@@ -196,7 +196,7 @@ main(argc, argv)
 #endif
 		fprintf(out, "H'%08x: ", my_pc);
 
-		my_opcode = char2short(&file[i]);
+		my_opcode = char2short((uint8_t*)&file[i]);
 		fprintf(out, "H'%04x  ", my_opcode);
 
 		for (j = 0; j < (N_O_BITS/8); j++) {
diff --git a/dcdis.h b/dcdis.h
index d385f3b..200317c 100644
--- a/dcdis.h
+++ b/dcdis.h
@@ -12,7 +12,7 @@
 #ifndef _DCDIS_H_
 #define _DCDIS_H_
 
-#include <sys/types.h>
+#include <stdint.h>
 
 #define N_O_BITS 16	/* SH-4 instructions are 16-bit fixed width */
 #define START_ADDRESS	0x8c010000
diff --git a/decode.c b/decode.c
index bc1039d..af16527 100644
--- a/decode.c
+++ b/decode.c
@@ -16,6 +16,7 @@
 #include "symtab.h"
 #endif
 #include <stdio.h>
+#include <string.h>
 
 extern int standard_disp;
 
@@ -50,7 +51,7 @@ decode(opcode, cur_PC, file, size, start_address)
 			reference = IMM*2+cur_PC+4;
 			reference2 = (reference&0x1fffffff)-(start_address&0x1fffffff);
 			if (reference2 >= 0 && reference2 <= size) {
-				literal = char2short(&file[reference2]);
+				literal = char2short((uint8_t*)&file[reference2]);
 			}
 #ifdef DO_SYMBOL
 			my_sym = (char *)symtab_lookup(reference);
@@ -80,7 +81,7 @@ decode(opcode, cur_PC, file, size, start_address)
 			reference = (IMM*4+(cur_PC&0xfffffffc)+4);
 			reference2 = (reference&0x1fffffff)-(start_address&0x1fffffff);
 			if (reference2 >= 0 && reference2 <= size) {
-				literal = char2int(&file[reference2]);
+				literal = char2int((uint8_t*)&file[reference2]);
 			}
 #ifdef DO_SYMBOL
 			my_sym = (char *)symtab_lookup(reference);
diff --git a/decode.h b/decode.h
index 48d01e0..639435f 100644
--- a/decode.h
+++ b/decode.h
@@ -12,7 +12,7 @@
 #ifndef _DECODE_H_
 #define _DECODE_H_
 
-#include <sys/types.h>
+#include <stdint.h>
 
 char *decode(uint16_t, uint32_t, char *, uint32_t, uint32_t);
 
diff --git a/symtab.c b/symtab.c
index 6186d00..e9c07fc 100644
--- a/symtab.c
+++ b/symtab.c
@@ -86,8 +86,8 @@ symtab_read(fp)
 	FILE *fp;
 {
 
-	unsigned char buf[256];
-	unsigned char buf2[256];
+	char buf[256];
+	char buf2[256];
 	char *ptr;
 	struct symtab *entry;
 	int i;
@@ -134,7 +134,7 @@ symtab_read(fp)
 			symtab_insert(entry);
 		}
 	}
-
+	return 0;
 }
 
 char *
diff --git a/symtab.h b/symtab.h
index 4dc82cc..29a72d3 100644
--- a/symtab.h
+++ b/symtab.h
@@ -12,7 +12,7 @@
 #ifndef _SYMTAB_H_
 #define _SYMTAB_H_
 
-#include <sys/types.h>
+#include <stdint.h>
 #include "dcdis.h"
 #include <stdio.h>
 
